framework/resource/retryresource/resource.go

```diff
@@ -38,13 +41,25 @@ func New(config Config) (*Resource, error) {
 		config.BackOff = backoff.NewExponentialBackOff()
 	}
 
+	var name string
+	{
+		u, err := internal.Underlying(config.Resource)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		name = u.Name()
+	}
```

Comment:
This here means the retry resource itself does not have any name. Why is that? Don't we use this name for anything? If not, I would like to call the internal `name` something like `wrappedName` or `underlyingName`, depending on the naming we want to apply now.

Reply:
It was never used. I told that already in my previous PR.
As you can see it was so confusing nobody noticed so far. `underlyingName` works for me.